spec/src/main/asciidoc/architecture.adoc

```diff
@@ -159,10 +181,13 @@ There is one `MetricRegistry` instance for each of the scopes listed in <<scopes
 Metrics can be added to or retrieved from the registry either using the `@Metric` annotation
 (see <<api-annotations, Metrics Annotations>>) or using the `MetricRegistry` object directly.
 
+A metric is uniquely identified by the `MetricRegistry` if the combination of the name and tags (order independent) in the Metadata are unique.
```

Comment:
Not sure about the wording of "order independent" - I think I would like to make it more explicit like what JMX ObjectName does for the key=value pairs